### DISCUSSION TITLE
Best practices for handling very high volume of logs

### DISCUSSION THREAD

**Discussion by bryceremick** [isar/isar#980] [Help]
Hello, 

In my use case, I need to store logs that are generated every 5-10ms. What do you think would be the best way to handle this using Isar?

Should I:
- Write each log individually as they are generated?
OR
- Batch logs in memory, and then write many with a single transaction?

Also, I'm not fully understanding the way this library works with isolates. Are all transactions executed on a separate isolate? Or do I need to implement my own isolate to do database writes off the UI thread? Do synchronous writes block the UI thread?

It is critical for me that writing the logs to disk does not block the UI(or completely fill up it's event loop).  

Thanks!

**Comment by JaffaKetchup** [comment]
Hi @bryceremick,
I recently ran into something similar myself, except my writes were every ~2-5ms. Batching is definitely the way to go here, opening transactions is just too expensive.
I implemented a long-lasting isolate that received a setup message then many objects that would be buffered into memory, until a null message, which signified to empty the buffer. In my case, I also cleared the buffer every time it got too large. If you would like, I can share some code :)

**Reply by bryceremick** [reply]
Thanks for your reply @JaffaKetchup. I ultimately ended up doing something similar to your approach. Only difference is that I write all my logs to the database after every X amount of logs. Not sure how much performance I gain from doing it in an Isolate though.. Will have to benchmark that.
